fpga_interchange/populate_chip_info.py

```diff
@@ -1740,6 +1760,426 @@ def get_cell(cell_type):
+                                   port.instance_name in cell_instances:
+                                    cell_name = port.instance_name
+                                    cell = cell_instances[cell_name].cell_name
+                                    pin = port.name
+                                    connection_type = 2 if cell_pin_direction_map[
+                                        cell][pin] == node_pin_dir else int(
+                                            node_pin_dir)
+                                    if cell_name not in connections_map[node]:
+                                        connections_map[node][cell_name] = []
+                                    connections_map[node][cell_name].append(
+                                        (connection_type, node_pin, pin, cell))
+        temp_value = {}
+        for node, _dict in cell_instances.items():
+            temp = []
+            temp_value[idx_map[node]] = {}
+            temp_value[idx_map[node]]['cell_type'] = _dict.cell_name
+            for connected_cell in connections_map[node].items():
+                temp.append((connected_cell[1], idx_map[connected_cell[0]]))
+            temp_value[idx_map[node]]['connections'] = temp
+            temp_value[idx_map[node]]['used_ports'] = used_ports[node]
+        ret_value = []
+        for idx, node in temp_value.items():
+            ret_value.append((idx, node['cell_type'], node['connections'],
+                              node['used_ports']))
+        ret_value.sort()
+        return ret_value, idx_map
+
+    prims = device.get_primitive_library()
+
+    clusters = []
+    if 'macros' in prims.libraries:
+        macro_lib = prims.libraries['macros']
+        cell_pin_direction_map = {}
+
+        macros_expansion = macro_lib.cells
+        primitives = prims.libraries['primitives'].cells
+        for cell_name, prim in primitives.items():
+            cell_pin_direction_map[cell_name] = {}
+            for port_name, port in prim.ports.items():
+                cell_pin_direction_map[cell_name][port_name] =\
+                    port.direction == Direction.Output
+
+        cell_site_type_map = {}
+        cell_site_type_to_bels_map = {}
+        cell_bel_site_type_to_bel_pins_map = {}
+        for mapping in device.yield_cell_bel_mappings():
+            if mapping.cell not in cell_site_type_map:
+                cell_site_type_map[mapping.cell] = set()
+            for site_type, bel in mapping.site_types_and_bels:
+                cell_site_type_map[mapping.cell].add(site_type)
+                if (mapping.cell, site_type) not in cell_site_type_to_bels_map:
+                    cell_site_type_to_bels_map[(mapping.cell, site_type)] = []
+                cell_site_type_to_bels_map[(mapping.cell,
+                                            site_type)].append(bel)
+
+        constants = device.get_constants()
+
+        for macro in macro_lib.cells.items():
```

Comment:
for readability, this should be sth like:

```
for macro_name, macro_data in macro_lib.cells.items():
```

rather than using later on `macro[0]` or `macro[1]`